src/main/java/com/xeeva/navigation/MainNav.java

```diff
@@ -1,317 +1,339 @@
-		btnCheckOut.syncVisible(5, false);
-		btnCheckOut.click();
-		//Wait for the cart pop up to close
-		btnsaveCartPopUp.syncHidden(10);
-	}
-
-
-	/**
-	 * @summary Method 'to click on Cart Check Out 
-	 * @author Lalitha Banda
-	 * @date 15/9/16
-	 **/
-	public void cart_CheckOut(){
-		clickCartItemsLink();
-		saveCartPopUp();
-		CheckOut();
-		
-	}
-}
+package com.xeeva.navigation;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import com.orasi.core.interfaces.Button;
+import com.orasi.core.interfaces.Element;
+import com.orasi.core.interfaces.Link;
+import com.orasi.core.interfaces.Listbox;
+import com.orasi.core.interfaces.Textbox;
+import com.orasi.core.interfaces.Webtable;
+import com.orasi.core.interfaces.impl.internal.ElementFactory;
+import com.orasi.utils.Constants;
+import com.orasi.utils.OrasiDriver;
+import com.orasi.utils.PageLoaded;
+import com.orasi.utils.Sleeper;
+import com.orasi.utils.TestReporter;
+
+public class MainNav {
+	PageLoaded pl = new PageLoaded();
+	private OrasiDriver driver = null;
+
+	/** Page Elements **/
+	@FindBy(linkText = "Logout") private Link lnkLogout;
+	@FindBy(xpath = ".//*[@value='Save Cart'][contains(@onclick, 'javascript')]")	private Button btnSaveCart;
+	@FindBy(id = "lnkShowPopup")	private Link lnkCartItem;
+	@FindBy(linkText = "tblCartInfo")	private Webtable cartTable;
+	@FindBy(xpath = ".//*[@id='tblCartInfo']/tbody/tr")	private List<WebElement> cartItemGridRows;
+	@FindBy(id = "txtQuantity")	private Textbox txtQuantity;
+	@FindBy(id = "ddlUOM")	private Listbox unitOfMeasure;
+	@FindBy(id = "txtUnitPrice")	private Textbox txtUnitPrice;
+	@FindBy(id = "btnSubmit")	private Button btnSubmit;
+	@FindBy(xpath = ".//*[@id='txtItem']")	private Textbox itemDescription;
+	@FindBy(xpath = "//span[@id='spanCartValue']")	private Element lblCartValue;
+
+
+	// Update Cart Items - for non price agreement cart Items verifications
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[2]/td[2]/input")
+	private Textbox txtItemQuantity;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[1]/td[4]/select")
+	private Listbox lstItemunitOfMeasure;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[1]/td[2]/input[1]")
+	private Textbox txtItemUnitPrice;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[2]/td[2]/a/div")
+	private Button btnAddToCart;
+	@FindBy(xpath = ".//*[@value='Save Cart'][contains(@onclick, 'javascript')]") private Button btnsaveCartPopUp;
+	@FindBy(xpath = "//*[@id='btnPopupCP']") private Button btnCheckOut;
+
+	/** Constructor **/
+
+	public MainNav(OrasiDriver driver) {
+		this.driver = driver;
+		pl.isDomComplete(driver);
+		ElementFactory.initElements(driver, this);
+	}
+
+	/** Page Interactions **/
+
+	public void pageLoaded() {
+		lnkLogout.syncVisible(10, false);
+	}
+
+	public boolean isLogoutDisplayed() {
+		try {
+			Thread.sleep(3000);
```

Comment:
Instead of doing a Thread.sleep surrounded by try/catch, let's use the Sleeper class instead.  So would be Sleeper.sleep(3000)